tests/testsuite/rustc_info_cache.rs

```diff
@@ -88,17 +88,108 @@ fn rustc_info_cache() {
         .env("CARGO_LOG", "cargo::util::rustc=debug")
         .env("RUSTC", other_rustc.display().to_string())
         .with_stderr_contains("[..]different compiler, creating new rustc info cache[..]")
-        .with_stderr_contains(miss)
-        .with_stderr_does_not_contain(hit)
-        .with_stderr_contains(update)
+        .with_stderr_contains(MISS)
+        .with_stderr_does_not_contain(HIT)
+        .with_stderr_contains(UPDATE)
         .run();
 
     p.cargo("build")
         .env("CARGO_LOG", "cargo::util::rustc=debug")
         .env("RUSTC", other_rustc.display().to_string())
         .with_stderr_contains("[..]reusing existing rustc info cache[..]")
-        .with_stderr_contains(hit)
-        .with_stderr_does_not_contain(miss)
-        .with_stderr_does_not_contain(update)
+        .with_stderr_contains(HIT)
+        .with_stderr_does_not_contain(MISS)
+        .with_stderr_does_not_contain(UPDATE)
         .run();
 }
+
+#[cargo_test]
+fn rustc_info_cache_with_wrappers() {
+    let wrapper_project = project()
+        .at("wrapper")
+        .file("Cargo.toml", &basic_bin_manifest("wrapper"))
+        .file("src/main.rs", r#"fn main() { }"#)
+        .build();
+    let wrapper = wrapper_project.bin("wrapper");
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "test"
+                version = "0.0.0"
+                authors = []
+                [workspace]
+            "#,
+        )
+        .file("src/main.rs", r#"fn main() { println!("hello"); }"#)
+        .build();
+
+    for &wrapper_env in [
+        "RUSTC_WRAPPER",
+        // FIXME: the test currently fails with RUSTC_WORKSPACE_WRAPPER, because
+        // cargo doesn't rebuild the package at all. That is, something in the
+        // fingerprint calculation is broken as well?
+        //
+        // "RUSTC_WORKSPACE_WRAPPER",
```

Comment:
I'll return to this later, but, I am wondering @ehuss if you know right of the bat if we indeed might not account for `workspace_wrapper` changes when deciding if we need to re-run the build?

Reply:
It looks like the [rustc-info cache](https://github.com/rust-lang/cargo/blob/57c512b3dfc47a6fc62d978313ec4eac7757f6de/src/cargo/core/compiler/build_context/target_info.rs#L142) uses [`process`](https://github.com/rust-lang/cargo/blob/57c512b3dfc47a6fc62d978313ec4eac7757f6de/src/cargo/util/rustc.rs#L87-L90) instead of [`workspace_process`](https://github.com/rust-lang/cargo/blob/57c512b3dfc47a6fc62d978313ec4eac7757f6de/src/cargo/util/rustc.rs#L92-L97) to determine the information. That means it will use `RUSTC_WRAPPER` but not `RUSTC_WORKSPACE_WRAPPER`.

I think it should be fine to change it to `workspace_process`. Cargo generally assumes that the wrapper will have the same behavior.  That should bust the cache and trigger an error the same.

Reply:
Yeah, that did the trick!